database/CoreDatabase.java

```diff
@@ -437,7 +444,15 @@ private void shutdownRocksPropertiesLogger() {
      */
     protected void closeResources() {
         sessions.values().forEach(p -> p.first().close());
-        statisticsBgCounterStop();
+        statisticsCorrector.close();
+        Optional<CompletableFuture<Void>> correction = statisticsCorrector.runningCorrection();
+        if (correction.isPresent()) {
+            try {
+                correction.get().get(Executors.SHUTDOWN_TIMEOUT_MS, MILLISECONDS);
+            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                LOG.warn("Statistics corrector was not fully shutdown");
+            }
+        }
```

Comment:
we try to safely shut down the corrector before killing RocksDB otherwise we can cause assertion errors in RocksDB